### DISCUSSION TITLE
Trying to call setAllFilters([]) and have an error

### DISCUSSION THREAD

**Discussion by danylboiko95** [TanStack/table#2253] [General]
Trying to filter by multiple values and have an error. Also, I tried to pass an array to setFilter, but it doesn't work

I can't find a way to filter column by setting multiple values. For example, I have a column where I have rows with such values: 'on progress', 'hold', 'unrated', 'done' and I want to see only rows where I have values 'on progress' and 'done'. How can I do it?

https://codesandbox.io/s/reverent-meadow-7fbrf

- Version 7.0.4

**Comment by JANA-afung** [comment]
The column object doesnt have setAllFilter().. Sounds like you want setFilter()

**Comment by danylboiko95** [comment]
> The column object doesnt have setAllFilter().. Sounds like you want setFilter()

Oh good. I tried to call it not only from a column. I ran into this problem when I was trying to set a filter on one column by several values. Perhaps you have some ideas?

**Comment by danylboiko95** [comment]
Put three examples in codesandbox https://codesandbox.io/s/reverent-meadow-7fbrf

**Comment by ivaskevych** [comment]
You need three things:
1. Correct `onChange `handler for your multiselect input
```JavaScript
function MultiSelectColumnFilter({
  column: { filterValue, setFilter, preFilteredRows, id },
}) {
  // Calculate the options for filtering
  // using the preFilteredRows
  const options = React.useMemo(() => {
    const options = new Set()
    preFilteredRows.forEach(row => {
      options.add(row.values[id])
    })
    return [...options.values()]
  }, [id, preFilteredRows])

  // Render a multi-select box
  return (
    <select
      multiple // this prop for multiselect
      value={filterValue}
      // an appropriate multiselect handler (use Ctrl key to select multiple)
      onChange={e => {
        const allValues = Array.from(
          e.target.selectedOptions,
        ).map(o => o.value).filter(Boolean)
        setFilter(allValues && allValues .length ? allValues : undefined);
      }}
    >
      <option value="">All</option>
      {options.map((option, i) => (
        <option key={i} value={option}>
          {option}
        </option>
      ))}
    </select>
  )
}
```
2. Filter type to handle multiple values filtering
  ```JavaScript
const filterTypes = React.useMemo(
    () => ({
      multiple: (rows, id, filterValue) => {
        return rows.filter(row => {
          const rowValue = row.values[id];
          return rowValue !== undefined
            ? filterValue.includes(rowValue)
            : true;
        });
      }
    }),
    []
  );
```
3. Use both of them for columns you need
 ```JavaScript
 const columns = React.useMemo(
    () => [
      {
        Header: "Info",
        columns: [
          {
            Header: "Status",
            accessor: "status",
            Filter: MultiSelectColumnFilter,
            filter: "multiple"
          }
        ]
      }
    ],
    []
  );
```

**Reply by lchelani** [reply]
what if we need to do multi select filter for the hidden columns?
